lake/api/api.go

```diff
@@ -44,6 +44,8 @@ type Interface interface {
 	UpdateIndex(ctx context.Context, names []string, pool ksuid.KSUID, branchName string) (ksuid.KSUID, error)
 	AddVectors(ctx context.Context, pool ksuid.KSUID, branch string, objects []ksuid.KSUID, message api.CommitMessage) (ksuid.KSUID, error)
 	DeleteVectors(ctx context.Context, pool ksuid.KSUID, branch string, objects []ksuid.KSUID, message api.CommitMessage) (ksuid.KSUID, error)
+	Vacuumable(ctx context.Context, pool, ref string) ([]ksuid.KSUID, error)
+	Vacuum(ctx context.Context, pool string, objects []ksuid.KSUID) ([]ksuid.KSUID, error)
```

Comment:
1. Yeah I was thinking about this over the weekend and thing we could do with just one endpoint with a dry run / delete flag. Will fix.
2. Yeah I think that makes sense will fix.

Reply:
In terms of the delete flag, I think the flag should be called "dryrun" since if the endpoint's name is vacuum it suggests that the default is that the object is deleted.

Reply:
@mattnibs: For the HTTP API, do you think GET /pool/{pool}/vacuum/{revision} for a dry run and POST with the same URL for deletion?

Reply:
Having thought about this far more than  warranted, I'm leaning towards keeping this all one endpoint. This is more based on intuition than anything but it seems that if Vacuum exists as only a single call in lake/api.Interface than it should also only have one endpoint.